armi/utils/asciimaps.py

```diff
@@ -40,236 +60,428 @@ class AsciiMap(object):
     """
     Base class for maps.
 
-    These should be able to read and write ASCII maps.
+    These should be able to read and write ASCII maps loaded either from text
+    or programatically with i,j / specifiers.
     """
 
-    def __init__(self, lattice=None):
-        self.lattice = lattice or {}
-
-    def readMap(self, text):
+    def __init__(self):
+        self.asciiLines = []
+        """A list of lines, each containing a list of ascii labels for each column. No blanks."""
+
+        self.asciiOffsets = []
+        """A list of offsets for each line that will be prepended before the contents of asciiLines"""
+
+        self.asciiLabelByIndices = {}
+        """A mapping from grid location objects to ascii labels"""
```

Comment:
is all of this instance state necessary? asciimaps are inherently ephemeral, so maybe these should be whipped up, passed to something that knows how to turn them into text/dict, and throw them away. food for thought. tbh, I don't see why these need to be classes at all, or maybe classes with almost all static methods. worth thinking about

Reply:
They're all things that need to be built up before the object can be used, and tehey can be built from two directions (from ascii vs. from data) so I kind of think they're useful here.